### PR TITLE
Add classes

### PR DESCRIPTION
**Pull request by abiola-arobieke**
In this milestone:

The person.rb, student.rb, and teacher.rb files were added to the project.
The person, student, and teacher classes were added to the files above.
Student and Teacher classes inherited the base class Person.
Class instance and constructor were added to each class.
Methods that can be accessed publicly were added to each class.
Private method was added to the person class.

### CODE REVIEW

File: classes/person.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  # accessor set methods
+  def of_age
+    @age >= 18
+  end
+
+  private :of_age
+
+  def can_use_services
+    of_age && @parent_permission
+  end
```
Comment (ShadyShawkat) [inline classes/person.rb]: - [x] Please make sure that this method is implemented correctly; this method should return `true` if either of the conditions is true (in other words not both of them have to be "true"), as per [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=if%20person%20is%20of%20age%20or%20if%20they%20have%20permission%20from%20parents.) requirement.

File: classes/student.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require './classes/person'
+
+# Define a sub-base class
+class Student < Person
+  def initialize(classroom, age)
+    super(age)
```
Comment (ShadyShawkat) [inline classes/student.rb]: - [x] Kindly make sure to call the `super` method correctly with the right parameters according to the parent's constructor. The `initialize` method here should accept all of the parameters to pass down to the `super` constructor.

```ruby
def initialize(age, classroom, name = "Unknown", parent_permission = true)
```

File: classes/teacher.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require './classes/person'
+
+# Define a sub-base class
+class Teacher < Person
+  def initialize(specialization, age)
+    super(age)
```
Comment (ShadyShawkat) [inline classes/teacher.rb]: - [ ] Kindly make sure to call the `super` method correctly with the right parameters according to the parent's constructor. The `initialize` method here should accept all of the parameters to pass down to the `super` constructor.

```ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
```

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @abiola-arobieke, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @abiola-arobieke , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Added Person, Student & Teacher classes ✔️
- Well Structured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**